lib/backend/k8s/k8s.go

```diff
@@ -79,12 +98,91 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
 	}
 
 	// Create the clientset
-	cs, err := clientset.NewForConfig(config)
+	cs, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
 	log.Debugf("Created k8s clientSet: %+v", cs)
-	return &KubeClient{clientSet: cs}, nil
+
+	tprClient, err := buildTPRClient(config)
+	if err != nil {
+		return nil, err
+	}
+	kubeClient := &KubeClient{
+		clientSet: cs,
+		tprClient: tprClient,
+	}
+
+	// Ensure the necessary ThirdPartyResources exist in the API.
+	err = kubeClient.ensureThirdPartyResources()
+	if err != nil {
+		return nil, goerrors.New(fmt.Sprintf("Failed to create necessary ThirdPartyResources: %s", err))
+	}
+
+	return kubeClient, nil
+}
+
+// TODO: The way this is right now, we'll run this every time
+// a client is created.  This is OK for felix, but less nice for say,
+// calicoctl.
+func (c *KubeClient) ensureThirdPartyResources() error {
+	// Ensure a resource exists for Calico global configuration.
+	tpr := extensions.ThirdPartyResource{
+		ObjectMeta: kapiv1.ObjectMeta{
+			Name:      "global-config.projectcalico.org",
+			Namespace: "kube-system",
+		},
+		Description: "Calico Global Configuration",
+		Versions:    []extensions.APIVersion{{Name: "v1"}},
+	}
+	_, err := c.clientSet.Extensions().ThirdPartyResources().Get(tpr.ObjectMeta.Name)
+	if err != nil {
+		// The resource needs to be created.
+		_, err := c.clientSet.Extensions().ThirdPartyResources().Create(&tpr)
```

Comment:
Are we "handling" any of the errors we get from the k8s client at the moment?  I think we should be looking at the returned error and converting a known error type for our API (in particular resource already exists, or resource not found etc) .... At the moment it looks as though k8s errors are passed straight up which means a user of the API has to understand k8s errors as well as etcd etc.  Perhaps not something to worry about for v1.0.0 - although I would prefer to see some error handling for that release.